code/binary-tree/src/test/java/ru/unn/agile/binaryTree/model/BinaryTreeManipulationTests.java

```diff
@@ -0,0 +1,53 @@
+package ru.unn.agile.binarytree.model;
+
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class BinaryTreeManipulationTests {
+    private BinaryTree binaryTree;
+
+    private int     simpleKey  = 42;
+    private String  simpleData = "SimpleText";
+
+    @Before
+    public void setUp() {
+        binaryTree = new BinaryTree();
+        binaryTree.add(simpleKey, simpleData);
+    }
+
+    @Test
+    public void canFind() {
+        final String result = binaryTree.find(simpleKey);
+
+        assertEquals(simpleData, result);
+    }
+
+    @Test
+    public void canRemove() {
+        binaryTree.remove(simpleKey);
+        final String result = binaryTree.find(simpleKey);
+
+        assertNull(result);
```

Comment:
The test name and its assertion is misleading. 
Assertion: cannot find removed element
Test name: can remove

Reply:
The next test is testing the same noErrorForFindNotExistsKey